model/project_ref.go

```diff
@@ -1892,19 +1892,23 @@ func (p *ProjectRef) UpdateAdminRoles(toAdd, toRemove []string) error {
 		adminUser, err := user.FindOneById(addedUser)
 		if err != nil {
 			catcher.Wrapf(err, "error finding user '%s'", addedUser)
+			p.removeFromAdmins(addedUser)
 			continue
 		}
 		if adminUser == nil {
 			catcher.Errorf("no user '%s' found", addedUser)
+			p.removeFromAdmins(addedUser)
 			continue
 		}
 		if err = adminUser.AddRole(role.ID); err != nil {
 			catcher.Wrapf(err, "error adding role %s to user %s", role.ID, addedUser)
+			p.removeFromAdmins(addedUser)
 			continue
 		}
 		if viewRole != "" {
 			if err = adminUser.AddRole(viewRole); err != nil {
 				catcher.Wrapf(err, "error adding role %s to user %s", viewRole, addedUser)
+				p.removeFromAdmins(addedUser)
```

Comment:
Hmm theoretically we shouldn't remove them here since they _did_ get the admin role successfully and that's the more important one.